pkg/rtc/room.go

```diff
@@ -902,17 +924,14 @@ func (r *Room) onTrackPublished(participant types.LocalParticipant, track types.
 	r.trackManager.AddTrack(track, participant.Identity(), participant.ID())
 
 	// launch jobs
+	r.lock.Lock()
 	_, hasPublished := r.hasPublished.Swap(participant.Identity(), true)
+	publisherAgentsEnabled := r.publisherAgentsEnabled.Load()
```

Comment:
Is there a race here since we are not setting r.publisherAgentsEnabled under this lock? ie, is it possible for onTrackPublished to be called twice?

Reply:
we are setting it under that lock

Reply:
But the lock is released between read and write. Is there anything else guaranteeing atomicity?

Do we also need an atomic here since it looks like it's always read and written under the lock?

Reply:
hasPublished is only set under lock, at the same time as publisherAgentsEnabled is checked. If publisherAgentsEnabled is set to true later, it's also locked, and anything in `hasPublished` is also read at that time

Reply:
I don't think either of them need to be atomic anymore